Tests/DatadogTests/Datadog/Mocks/ServerMock.swift

```diff
@@ -226,3 +226,20 @@ extension ServerMock {
         }
     }
 }
+
+// MARK: - RUM feature helpers
+
+extension ServerMock {
+    func waitAndReturnRUMEventMatchers(count: UInt, file: StaticString = #file, line: UInt = #line) throws -> [RUMEventMatcher] {
```

Comment:
if we had a similar method for other mathcers, is it possible to make this generic function?

Reply:
IMO it will be more hassle to work with generics and no gain. Instead of this usage:
```swift
let logMatchers = serverMock.waitAndReturnLogMatchers(count: 2)
let spanMatchers = serverMock.waitAndReturnSpanMatchers(count: 2)
let rumEventMatchers = serverMock.waitAndReturnRUMEventMatchers(count: 2)
```
we'd be typing sth like this in every place:
```swift
let logMatchers = serverMock.waitAndReturnMatchers(count: 2, genericKind: LogMatcher.self)
let spanMatchers = serverMock.waitAndReturnMatchers(count: 2, genericKind: LogMatcher.self)
let rumEventMatchers = serverMock.waitAndReturnMatchers(count: 2, genericKind: LogMatcher.self)
```
the `genericKind` would have to let the `ServerMock` map the request payload `Data` to a concrete matcher. And this mapping is different for Logs and Spans + RUM Events, as they use different payload format.

There's almost no code duplication between 3 existing methods.

Reply:
you don't need `genericKind` parameter, you can do `let logMatchers: [LogMatcher] = serverMock.waitAndReturn...`
if it doesn't make life easier, nevermind then 👌

Reply:
The thing is that all 3 matchers offer also other factory methods:
```swift
LogMatcher.fromArrayOfJSONObjectsData(Data) -> [LogMatcher]
LogMatcher.fromJSONObjectData(Data) -> LogMatcher

SpanMatcher.fromNewlineSeparatedJSONObjectsData(Data) -> [SpanMatcher]
SpanMatcher.fromJSONObjectData(Data) -> SpanMatcher

RUMEventMatcher.fromNewlineSeparatedJSONObjectsData(Data) -> [RUMEventMatcher]
RUMEventMatcher.fromJSONObjectData(Data) -> RUMEventMatcher // not yet existing, but may come later
```

Finding a common name for all three will IMO immerse the readability:
```swift
Matcher.fromMultiObjectPayload(Data) -> [Matcher]
Matcher.fromSingleiObjectPayload(Data) -> Matcher
```
plus, it will require quite a refactoring effort with no much gain.